### ISSUE TITLE
`@interval(3,2)` not throwing an ArgumentError

### ISSUE THREAD

**Issue by lbenet** [dpsanders/ValidatedNumerics.jl#127]
Rereading [this comment](https://github.com/dpsanders/ValidatedNumerics.jl/pull/55#issuecomment-139120253), I just realized that `@interval` and `@decorated` are not behaving as they should. For example:
```julia
julia> using ValidatedNumerics

julia> displaymode(format=:full, decorations=true)

julia> @interval(3,2)  # should yields an error!
Interval(2.0, 3.0)

julia> @decorated(3,2) # should yield nai with ill-decoration
DecoratedInterval(Interval(2.0, 3.0), com)
```

This is in contrast with the use of `Interval` or `DecoratedInterval` directly.
```julia
julia> Interval(3,2)
ERROR: ArgumentError: Must have a ≤ b to construct Interval(a, b).
 in call at /Users/benet/.julia/v0.4/ValidatedNumerics/src/intervals/intervals.jl:19
 in call at /Users/benet/.julia/v0.4/ValidatedNumerics/src/intervals/intervals.jl:36

julia> DecoratedInterval(3,2)
ERROR: ArgumentError: Must have a ≤ b to construct Interval(a, b).
 in call at /Users/benet/.julia/v0.4/ValidatedNumerics/src/intervals/intervals.jl:19
 in call at /Users/benet/.julia/v0.4/ValidatedNumerics/src/decorations/intervals.jl:51
```

The result is because the use of `hull` in the macros. I have no idea how to make that this behaves correctly.

**Comment by dpsanders** [comment]
Good point. Instead of hull we can just use a.lo and b.hi.   I'll do this later when I have a chance.

**Comment by lbenet** [comment]
I just played with your idea, but got stock elsewhere...

**Comment by dpsanders** [comment]
Where is it getting stuck? I am not at a computer right now.

**Comment by lbenet** [comment]
I'm adapting the tests; the problem was the way I was modifying the macro.

**Comment by dpsanders** [comment]
Closed by #128.
